pkg/provision/engine.go

```diff
@@ -135,6 +167,8 @@ func (e *defaultEngine) decommission(ctx context.Context, r *Reservation) error
 		return errors.Wrap(err, "decommissioning of reservation failed")
 	}
 
+	e.stats[r.Type].Decrement()
```

Comment:
This can/will go below zero (minus value) when we process expired, or to-delete reservations that were never provisioned on the node. There is no way (I know of) to check if the reservation to delete is active, or was never deployed.

Reply:
Well, at line 172, we remove the reservation from the reservation cache.
I think this cache is the most accurate view on what is actually deployed on the node or not. Maybe we could use it to return the number of different workload deployed at any point in time.
This kind of information is something I would also like to send to BCDB at some point.